servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletBuilder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServlet;
+
+import com.linecorp.armeria.server.ServerBuilder;
+
+/**
+ * A builder class which creates a new {@link ServletContextImpl} instance.
+ */
+public class ServletBuilder {
+
+    @Nullable
+    private ServletContextImpl servletContext;
+
+    private ServerBuilder serverBuilder;
+
+    /**
+     * Constructor.
+     */
+    public ServletBuilder(ServerBuilder serverBuilder) {
+        servletContext = new ServletContextImpl();
+        this.serverBuilder = serverBuilder;
+    }
+
+    /**
+     * Creates a new {@link DefaultServletService}.
+     */
+    public ServerBuilder build() {
+        return serverBuilder;
+    }
+
+    /**
+     * Add a servlet of the {@link DefaultServletService}.
+     */
+    public ServletBuilder addServlet(String path, HttpServlet httpServlet) {
+        final ServletRegistration.Dynamic sd = servletContext.addServlet(path, httpServlet);
```

Comment:
This have to be **Dynamic interface**, because **ServletRegistration interface** don't include `setLoadOnStartup()` method and `setAsyncSupported()` method.

Reply:
Yes, but it seems like we don't set `setAsyncSupported` as `true` for now. Also `setLoadOnStartup(1)` doesn't seem much meaning.
How about just using `ServletRegistration` here and
adding another method for async such as `addAsyncServlet` later?

Reply:
Agree, Armeria has been async already. We don't need **servlet async** sound like that add more queue + executor thread pool. In this case **servlet async** make system more complicated and not good performance.